README.md
# Astro

### A design system by Magnetis 👩‍🚀

[![Build Status](https://travis-ci.org/magnetis/astro.svg?branch=main)](https://travis-ci.org/magnetis/astro)

## Table of contents

- [Introduction](#introduction)
- [Supported libraries](#supported-libraries)
- [Adding Astro to your project](#adding-astro-to-your-project)
- [Using Astro](#using-astro)
- [Contributing](#contributing)
- [Troubleshooting](#troubleshooting)
- [Contact us](#contact-us)

## Introduction

Made for **Magnetis**' designers, developers, product managers, data analysts, data scientists, stakeholders and enthusiasts, **Astro** helps us protect the brand concepts, elevate consistency of user experience and increase the speed and efficiency of how we design and build products for our customers.

We believe that a good user experience is built with **consistent design**, well defined **visual and interaction patterns**, carefully built **emotions** and **made for all**.
Therefore, Astro is essentially:

### Atomic

Astro is built based on **Atomic Design**, so our designers and developers can use its style set and components to enhance our product interface, as a **modular system** that can be easily **managed and updated**.

### Open

Magnetis believes that good design and experiences can **impact the product market and society itself**. Our **design and code assets are open** so we can contribute to the community and help new product teams.

### Accessible

Design is meant to be **for all**. Astro style and components follow basic recommendations from the **WCAG guidelines** for accessibility, with tested contrast ratios and good practices in legibility and screen reading solutions.
[See WCAG >>](https://www.w3.org/TR/WCAG21/)

## Supported libraries

For now, Astro only supports React as its official library.

## Adding Astro to your project

### Installing the dependency

Install Astro via terminal:
Run `yarn add @magnetis/astro` or `npm install @magnetis/astro`.

### Using via CDN

```html
<link
  href="https://unpkg.com/@magnetis/astro/dist/astro.css"
  rel="stylesheet"
  type="text/css"
/>
```

## Using Astro

Ready to start coding with Astro? Follow the instructions in the docs ([astro.magnetis.com.br](https://astro.magnetis.com.br/)) to learn how to use our components. You'll also see how to apply attribute modifiers (such as classes) to your HTML elements and React components.

Here's a quick example. If you'd like to create a Display text styled in Astro typography:

1. Complete all install steps listed above;

2. Import Astro in your file through one of the following methods:

   - Inside the `<head>` tag:
     ```html
     <head>
       <link
         href="/node_modules/@magnetis/astro/dist/astro.css"
         rel="stylesheet"
         type="text/css"
       />
     </head>
     ```
   - With ES6 modules: `import "@magnetis/astro";`

   - With commonjs: `require("@magnetis/astro");`

3. Go to Astro's [Typography page](https://astro.magnetis.com.br/typography);

4. See that the "Display" section guides you to create a `<p>` element with a specific text display class;

5. Replicate the instructions in your page file and make sure the correct styles have been applied;

6. If your styles aren't rendered correctly, make sure Astro has been successfully installed into your project tree and `astro.css` is correctly imported.

Although our CSS variables _are_ available in the bundle, it's important to emphasize that you should use Astro classes whenever possible, instead of using the variables directly. Always look for the element you're creating in the docs before you start to build it from scratch.

In other words, avoid this:

```css
.paragraph {
  font-family: var(--font-primary);
  font-weight: 600;
  font-size: 24px;
  line-height: 1.5;
  color: var(--color-moon-900);
}
```

```html
<p class="paragraph">Don't do this</p>
```

In the example above, all these properties could be replaced with using the `a-text--large` class in the `paragraph` element. Like so:

```html
<p class="a-text--large">Do this!</p>
```

### Customizing styles

In case you absolutely have to customize one or more properties in an Astro component, you should still use the corresponding Astro class that's closer to your goal and override it with your custom properties. You can do this:

```css
.my-paragraph {
  color: var(--color-moon-700); /* override default Astro color */
}
```

```html
<p class="a-text--large my-paragraph">Customize like this</p>
```

## Contributing

It's awesome that you want to contribute to Astro! Please see [CONTRIBUTING.md](CONTRIBUTING.md) to learn how it works.

## Troubleshooting

- If you copy the markup from an example in our documentation, don't forget to change the attributes "className" to "class" and "htmlFor" to "for" if you're not in a React environment like our docs are.

## Contact us

Found a bug? Please see if it was already reported on our [issues page](https://github.com/magnetis/astro/issues). If you can't find a matching report, you can [open a new issue](https://github.com/magnetis/astro/issues/new) including as much information as you can gather. We appreciate it!

Any lingering questions? You can contact us at [dev@magnetis.com.br](mailto:dev@magnetis.com.br).

Thank you for using Astro! 🎉


RELEASES.md
## Release Checklist (Magnetis contributors only)

The Magnetis Astro team launches a new release every time we decide the latest changes should go live. When that occurs, we update the NPM Registry package and deploy the latest version of the documentation.

If you're a Magnetis employee who's about to launch a new release, please follow this checklist.

1. Create a new branch.
2. Update the CHANGELOG.md file. To make sure you add all changes, see the commit history: https://github.com/magnetis/astro/commits/main.
3. Bump the version directly on `package.json` file, following the [Semantic Versioning](https://semver.org/) convention.
4. Build the project: `yarn build`.
5. Push the branch to GitHub and open a PR (no need to ask for reviewers).
6. Merge with main.
7. Create a new release on GitHub: https://github.com/magnetis/astro/releases.
8. Send the release to NPM registry: `git checkout main` && `git pull` && `yarn publish` (you will be prompted about the version, just type in the same version you manually bumped).
9. Deploy to the production documentation: `yarn docs:publish`.
10. Notify about the release on #astro and #codehood channels on Slack.


doczrc.js
import { css } from 'docz-plugin-css';

const selectors = {
  wrapper: '& .css-pw2o4y',
  main: '& .css-1i0fr1k',
  logo: '& .css-c9p757',
  menuArea: '& .css-wwgiek',
  innerMenuArea: '& .css-pccrlp',
  menuItem: '& .css-82rrzo, .css-82rrzo.active, .css-82rrzo:visited',
  menuSubitemsArea: '& .css-1qc5yjq',
  menuSubItem: '& .css-1l1jc95, .css-1l1jc95:visited',
  menuBtn: '& .css-6pue65, .css-15exo54',
  menuBtnShape: '& .css-14xo6fr, .css-1v3qlja',
  mobileLogoArea: '& .css-12xm67f',
  mobileMenu: '& .css-k33w14',

  /* Docz default elements (hidden) */

  editPageBtn: '& .css-p4m8gg',
  githubBtn: 'a[aria-label="View source on Github"]',
  builtDocz: '& .css-b4nn6f',
  search: '& .css-1pxi6pq',
  menuBlueTop: '& .css-y1r3ra:before',
  menuItemsBorder: '& .css-1ccg7te:after',
  menuSubitemsBorder: '& .css-1evit71:after'
};

export default {
  hashRouter: true,
  base: '/',
  plugins: [
    css({
      preprocessor: 'postcss' /* stylelint-disable-line */
    })
  ],
  modifyBabelRc: babelrc => ({
    ...babelrc,
    plugins: [...babelrc.plugins, 'babel-plugin-inline-import']
  }),
  indexHtml: 'public/docz.html',
  htmlContext: {
    favicon: 'public/favicon.png'
  },
  themeConfig: {
    /* In this file, the font shorthand with variable only works if you'll use it whole don't need to alter any other font properties */
    styles: {
      body: {
        /* Hide Docz default elements */

        [selectors.editPageBtn]: {
          display: 'none'
        },
        [selectors.githubBtn]: {
          display: 'none'
        },
        [selectors.builtDocz]: {
          display: 'none'
        },
        [selectors.search]: {
          display: 'none'
        },
        [selectors.menuBlueTop]: {
          display: 'none'
        },
        [selectors.menuItemsBorder]: {
          display: 'none'
        },
        [selectors.menuSubitemsBorder]: {
          display: 'none'
        },

        /* Custom theme styles */

        [selectors.wrapper]: {
          /* Gradient header */
          backgroundImage: 'var(--gradient-nebulosa)',
          backgroundPosition: 'top',
          backgroundRepeat: 'no-repeat',
          backgroundSize: '100% 312px'
        },
        '@media (min-width: 768px)': {
          [selectors.main]: {
            padding: '0 40px 40px 100px'
          }
        },
        [selectors.logo]: {
          height: 171,
          width: 151,
          margin: '46px 60px',
          backgroundImage: 'url("public/astro-logo.svg")',
          backgroundSize: 164,
          backgroundPosition: 'center',
          backgroundRepeat: 'no-repeat',
          textIndent: -999
        },
        '& p': {
          font: 'var(--font-secondary)',
          color: 'var(--color-moon-900)!important'
        },
        a: {
          color: 'var(--color-uranus-700)'
        },
        [selectors.menuArea]: {
          backgroundColor: 'var(--color-space-100)',
          borderRight: '1px solid var(--color-space-300)',
          width: 320,
          [selectors.innerMenuArea]: {
            width: 320
          }
        },
        [selectors.menuItem]: {
          fontFamily: "'Poppins', sans-serif",
          fontSize: 20,
          fontWeight: 500,
          color: 'var(--color-moon-900)',
          borderTop: '1px solid var(--color-space-300)',
          padding: '22px 50px',
          '&:hover': {
            color: 'var(--color-moon-300)',
            fontWeight: 500
          },
          '&:after': {
            position: 'absolute',
            right: 32,
            width: 32,
            height: 32,
            content: '""',
            maskImage: 'url("public/drop-down.svg")',
            backgroundColor: 'var(--color-moon-900)'
          },
          '&.active:after': {
            maskImage: 'url("public/drop-up.svg")',
            backgroundColor: 'var(--color-moon-900)'
          }
        },
        '@media (max-width: 1119px)': {
          [selectors.menuItem]: {
            padding: '22px 39px'
          }
        },
        [selectors.menuSubitemsArea]: {
          margin: '5px 0 15px'
        },
        [selectors.menuSubItem]: {
          fontFamily: "'Poppins', sans-serif",
          fontSize: 16,
          fontWeight: 500,
          color: 'var(--color-moon-400)',
          padding: '6px 50px',
          opacity: 1,
          '&:hover': {
            color: 'var(--color-moon-300)'
          },
          '&.active': {
            color: 'var(--color-moon-900)'
          },
          '&:before': {
            display: 'none'
          }
        },
        [selectors.menuBtn]: {
          backgroundColor: 'transparent',
          '&:before': {
            display: 'none'
          }
        },
        [selectors.menuBtnShape]: {
          height: 0,
          border: '1px solid var(--color-space-100)',
          borderRadius: 3,
          backgroundColor: 'var(--color-space-100)'
        },
        [selectors.mobileLogoArea]: {
          padding: 0,
          '&:before': {
            display: 'none'
          }
        },
        [selectors.mobileMenu]: {
          backgroundColor: 'var(--color-space-100)'
        },
        '@media (max-width: 560px)': {
          '& h1.css-bdhje3': {
            /* Specificity hack for responsive h1 */
            fontSize: 48
          }
        }
      },
      h1: {
        fontFamily: "'Poppins', sans-serif",
        fontSize: 64,
        fontWeight: 600,
        letterSpacing: 'normal',
        color: 'var(--color-space-100)',
        margin: '103px 0 186px',
        ':before': {
          /* Removes bottom blue line */
          display: 'none'
        }
      },
      h2: {
        fontFamily: "'Poppins', sans-serif",
        fontSize: 24,
        fontWeight: 600,
        color: 'var(--color-moon-500)',
        borderBottom: 'none',
        margin: '0 0 15px',
        '& a[href]': {
          /* Selects download links in Typography page */
          color: 'var(--color-uranus-700)',
          textDecoration: 'underline',
          '&:hover': {
            color: 'var(--color-uranus-500)'
          },
          '&:active': {
            color: 'var(--color-uranus-800)'
          }
        },
        '&#manifesto, &#get-started': {
          /* Selects specific Homepage h2 headings */
          fontSize: 40,
          color: 'var(--color-moon-900)',
          marginTop: 64
        }
      },
      h3: {
        fontFamily: "'Poppins', sans-serif",
        fontSize: 16,
        fontWeight: 600,
        color: 'var(--color-moon-500)'
      },
      h4: {
        fontFamily: "'Lato', sans-serif",
        fontSize: 22,
        fontWeight: 400,
        color: 'var(--color-moon-400)',
        marginBottom: 65
      },
      playground: {
        padding: 16,
        overflowX: 'scroll',
        '& > *': {
          boxSizing: 'border-box',
          margin: 12
        },
        '& > .a-slider': {
          marginBottom: 22
        },
        '&.two-columns': {
          display: 'inline-grid',
          gridTemplateColumns: 'auto auto'
        }
      }
    }
  }
};
